### PR TITLE
Added new Forked and Star count feature

### PR DESCRIPTION
**Pull request by Ashutosh0602**
close: #320 

This new feature will tell the new contributors the size of community and will encourage them to contribute and actively participate more in improving the CARE.

It will show the combine counts of forked and start count of all repositories of CARE at once.

Screenshot 2024-03-04 at 2 14 05 PM

### CODE REVIEW

File: app/ForkedStar.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    `https://api.github.com/orgs/${env.NEXT_PUBLIC_GITHUB_ORG}/repos`,
+  );
+  const posts = await res.json();
+
+  return {
+    props: {
+      posts,
+    },
+  };
+}
```
Comment (rithviknishad) [inline app/ForkedStar.tsx:70]: This does not work in app router.

**Comment by Ashutosh0602** [comment]
@rithviknishad Is this okay or do you want to suggest some changes

File: app/ForkedStar.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  data?.map((ls: any) => {
+    if (ls.forks) {
+      forkCount += ls.forks;
+    }
+    if (ls.stargazers_count) {
+      starCount += ls.stargazers_count;
+    }
+    // console.log(ls.full_name, ls.forks, ls.stargazers_count);
+  });
+  console.log(forkCount);
```
Comment (rithviknishad) [inline app/ForkedStar.tsx:29]: Let's make this as Server component instead of a client component

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: app/ForkedStar.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+  return (
+    <div className="flex justify-center">
+      {forkCount > 0 && starCount > 0 ? (
+        <>
+          <div className="flex items-center justify-center bg-violet-950 m-4 p-2 w-6/12 rounded-md border-2 border-slate-400 ">
```
Comment (rithviknishad) [inline app/ForkedStar.tsx:35]: Let's not use colors outside `secondary` / `primary`. Different organizations deploying leaderboard would have different themes, and the CSS Variables secondary and primary uses the intended theme colors.

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: app/ForkedStar.tsx (hunk 2 above)
Comment (Ashutosh0602) [reply app/ForkedStar.tsx:29]: Okay so it means I can use getServerSideProps to make it server sider rendering?

Comment (rithviknishad) [reply app/ForkedStar.tsx:29]: Those are relevant to pages router only. These are app router pages.

Check out NextJS docs on App Router
https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating#fetching-data-on-the-server-with-fetch

**Comment by Ashutosh0602** [comment]
@rithviknishad I have made changes please review it.

File: app/ForkedStar.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        <></>
+      )}
+    </div>
+  );
+}
+
+async function getData() {
+  const res = await fetch(
+    `https://api.github.com/orgs/${env.NEXT_PUBLIC_GITHUB_ORG}/repos`,
+  );
```
Comment (rithviknishad) [inline app/ForkedStar.tsx:50]: 1. This fetches only first 30 repositories. 
2. Use the pre-authenticated `octokit` instance to make calls with GitHub APIs.
3. Use `octokit.paginate` to fetch all repositories from all pages.

File: app/ForkedStar.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+import { GoRepoForked } from "react-icons/go";
+import { GoStar } from "react-icons/go";
+
+export default async function ForkedStar() {
+  let forkCount = 0;
+  let starCount = 0;
+
+  const data = await getData();
+
+  data?.map((ls: any) => {
```
Comment (rithviknishad) [inline app/ForkedStar.tsx:11]: Avoid usage of `any` types. This won't be needed as octokit responses are typesafe.

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Pull request closed without merging**